Add TF 1.14 documentation for registry and fix up some other provider-related docs (#853)
hashicorp/web-unified-docs: content/terraform-docs-common/docs/registry/providers/docs.mdx

| `docs/`                     | `index.md`                | Index page for the provider.                                                                    |
| `docs/guides/`              | `<guide>.md`              | Additional documentation for guides.                                                            |
| `docs/resources/`           | `<resource>.md`           | Information for a Resource. Filename should not include a `<PROVIDER NAME>_` prefix.            |
| `docs/data-sources/`        | `<data_source>.md`        | Information on a provider data source.                                                          |
| `docs/functions/`           | `<function>.md`           | Information on a provider function.                                                             |
| `docs/ephemeral-resources/` | `<ephemeral-resource>.md` | Information for an Ephemeral Resource. Filename should not include a `<PROVIDER NAME>_` prefix. |
| `docs/actions/`             | `<action>.md`             | Information for an action. Filename should not include a `<PROVIDER NAME>_` prefix.             |

> | `docs/list-resources/`      | `<list-resource>.md`      | Information for a List Resource. Filename should not include a `<PROVIDER NAME>_` prefix.       |

Reviewer: ```suggestion
| `docs/list-resources/`      | `<list-resource>.md`      | Information for a list resource. Filename should not include a `<PROVIDER NAME>_` prefix.       |
```

* List any arguments for the provider block.
````

> #### Resource Type (managed, data source, ephemeral, etc.) Headers

Reviewer: ```suggestion
#### Resource Type Headers

Resource types include managed, data source, ephemeral, and others. 
```
Refer to the style guide about ["etc."](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/top-12.md#examples-8) and [parentheses](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/general/grammar.md#do-not-use-parentheses-en-dashes-or-em-dashes-to-separate-ideas-or-phrases)

## Navigation Hierarchy

> Provider docs are organized by category: functions, actions, list resources, ephemeral resources, resources, data sources, and guides. At a minimum, a provider must contain an index (`docs/index.md`) and at least one function, action, ephemeral resource, list resource, resource, or data source.

Reviewer: ```suggestion
Provider docs are organized by the following categories: 

- functions 
- actions
- list resources
- ephemeral resources 
- resources 
- data sources 
- guides 

At a minimum, a provider must contain an index file, `docs/index.md`, and at least one function, action, ephemeral resource, list resource, resource, or data source.
```

Refer to the [style guide](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/general/grammar.md#use-colons-to-introduce-lists-tables-and-visual-aids)
